src/e2e-tests/proxy_healthcheck_test.go

```diff
@@ -0,0 +1,128 @@
+package e2e_tests
+
+import (
+	"fmt"
+	"slices"
+	"time"
+
+	"e2e-tests/utilities/bosh"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Proxy healthcheck", Ordered, Label("proxy", "healthchecks"), func() {
+	var (
+		deploymentName string
+		proxyIPs       []string
+		proxies        []bosh.Instance
+		numProxies     int
+		err            error
+	)
+
+	BeforeAll(func() {
+		deploymentName = "pxc-proxy-healthcheck-" + uuid.New().String()
+
+		Expect(bosh.DeployPXC(deploymentName,
+			bosh.Operation(`use-clustered.yml`),
+			bosh.Operation(`iaas/cluster.yml`),
+			bosh.Operation(`require-tls.yml`),
+		)).To(Succeed())
+
+		proxies, err = bosh.Instances(deploymentName, bosh.MatchByInstanceGroup("proxy"))
+		Expect(err).NotTo(HaveOccurred())
+		proxyIPs, err = bosh.InstanceIPs(deploymentName, bosh.MatchByInstanceGroup("proxy"))
+		Expect(err).NotTo(HaveOccurred())
+		numProxies = len(proxyIPs)
+	})
+
+	AfterAll(func() {
```

Comment:
Nit:  With modern Ginkgo v2, I would prefer a DeferCleanup to an AfterAll node in most cases.  This lets you put the cleanup closer to where the allocation happened.  (e.g. immediately after `DeployPXC`).

Reply:
✔️